cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+
+	// TODO: add podman remote support
+
+	remoteFiles := []string{}
+	dials := []*ssh.Client{}
+	if scpOpts.ToRemote || scpOpts.FromRemote {
+		dials, remoteFiles, err = createConnections(uri, iden)
+		if err != nil {
+			return err
+		}
+	}
+
+	switch {
+	case scpOpts.FromRemote: // if we want to load FROM the remote
+		//	_, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.DestImageName, "", false)
+		err = fromRemote(dials[0], scpOpts.DestImageName, remoteFiles[0], scpOpts.Save.Output, "")
+		if err != nil {
+			return err
+		}
+		if scpOpts.ToRemote { // we want to load remote -> remote
+			//rep, err := execCommand(iden[1], uri[1], scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.Tag, true) // validate connection given and execute save or load on remote machine
+			rep, err := toRemote(dials[1], remoteFiles[1], scpOpts.Save.Output, "")
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+			break
+		}
+		report, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+	case scpOpts.ToRemote: // remote host load
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.DestImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		//rep, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.Load.Input, "", scpOpts.ToRemote)
+		rep, err := toRemote(dials[0], remoteFiles[0], scpOpts.Save.Output, "")
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	// TODO: Add podman remote support
+	default: // else native load
+		if scpOpts.Tag != "" {
+			return errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+		}
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.DestImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// toRemote takes a dialed ssh client, and image information. it connects to the client
+// and copies the saved image dir over to the remote host and then loads it onto the machine
+// returns a string containing output or an error
+func toRemote(dial *ssh.Client, remoteFile string, localFile string, tag string) (string, error) {
+	f, err := os.Open(localFile) // open image file to copy
```

Comment:
From https://github.com/containers/podman/pull/10828#discussion_r673286381 :

Nothing uses this AFAICS.